mimic/test/test_maas.py

```diff
@@ -1135,6 +1135,35 @@ def test_agenthostinfo_missing_include_returns_400(self):
         self.assertEquals(resp.code, 400)
         self.assertEquals(data['message'], 'Validation error for key \'include\'')
 
+    def test_view_connections_missing_agentid_400s(self):
+        """
+        Attempting to view connections without an agentId causes a 400.
+        """
+        (resp, data) = self.successResultOf(
+            json_request(self, self.root, b"GET",
+                         '{0}/views/connections'.format(self.uri)))
+        self.assertEquals(resp.code, 400)
+        self.assertEquals(data['message'], 'Validation error for key \'agentId\'')
+
+    def test_view_connections(self):
+        """
+        An existing agent returns a list of connections, and a non-existing agent
+        returns an empty list.
```

Comment:
It strikes me as odd that "no agent ID" is a 400, but "invalid agent ID" is a 200.  Is this the behavior of the actual upstream API?